src/flagStore.ts

```diff
@@ -114,6 +116,27 @@ export class FlagStore {
 		};
 	}
 
+	getFeatureFlag(key: string): Promise<FlagWithConfiguration | null> {
+		let flag = this.flagMetadata[key];
+		return new Promise((resolve, reject) => {
+			this.store.get(DATA_KIND, key, async (res: FlagConfiguration) => {
+				if (!res) {
+					resolve(null);
+				}
+				if (!flag || flag.environments[this.config.env].version < res.version) {
```

Comment:
Caching mechanism - only retrieve the feature flag if the environment version is out of date.